### DISCUSSION TITLE
Excluding tests with filters from CLI

### DISCUSSION THREAD

**Discussion by borislavml** [thomhurst/TUnit#829] [Q&A] [answered]
Is it possible to exclude some tests from running on the command line using attributes. With XUnit I can add an attribute like  say `[Trait("Category", "Integration")]` and then I can filter the tests wen running like this  `dotnet test  --filter "Category!=Integration"`. What would be the best way to do this with TUnit? I can see a `Category `attribute available but I can't find how to use it and if it's even applicable for filtering.

**Comment by thomhurst** [comment]
`dotnet run --treenode-filter /**[Category!=Integration]`

**Comment by thomhurst** [comment]
Filter info is here: https://thomhurst.github.io/TUnit/docs/tutorial-extras/test-filters

**Reply by borislavml** [reply]
So I get it that this is supported only through` dotnet run `and not` dotnet test`?

**Reply by thomhurst** [reply, marked as answer]
This should work: `dotnet test -- --treenode-filter /**[Category!=Integration]`

**Reply by borislavml** [reply]
Thank you, it works indeed but I have a scenario where it wouldn't work because of my setup use-case. I may post about see if it's possible to do what I want, but it's not TUnit issue for sure.

**Comment by andrej733** [comment]
Hello, will be this Category attribute shown in Test Explorer (Visual Studio)? When I enable traits in test explorer I still can't see it. I would like be able select only category which should be run. Also when writing code and just right click on method in which I am and choose Run Tests  it just build project but it not run them

**Reply by thomhurst** [reply]
> Hello, will be this Category attribute shown in Test Explorer (Visual Studio)? When I enable traits in test explorer I still can't see it. I would like be able select only category which should be run. Also when writing code and just right click on method in which I am and choose Run Tests it just build project but it not run them

Hmm I haven't done anything regarding this specifically. I don't even know if it's supported in the new Microsoft testing platform yet. I'll check

**Comment by borislavml** [comment]
@thomhurst I'm kind of stuck on this. I have CI/CD which runs `dotnet test {sln}`. The solution contains a bunch of xUnit tests and now I have added my TUnit project to that sln. So when I run the **dotnet test** command I now have to "filter twice", once using **--filter** for xUnit and once using **--treenode-filter** for TUnit but it seems impossible to supply both on the cli. I know this is not a problem of TUnit but it looks like a use-case which other people might have. So I'd appreciate your thoughts on this.

**Reply by borislavml** [reply]
I'm trying to run smth similar to
` dotnet test .\PMSProxy.sln --filter 'FullyQualifiedName!~Integration.Test' -- --treenode-filter /**[Category!=Integration]`

**Reply by thomhurst** [reply]
I don't think you can mix running xUnit and TUnit as xUnit uses VSTest and TUnit uses Microsoft.testing.platform

I would just run dotnet against the projects rather than the solution

**Reply by borislavml** [reply]
Yeah, I guess I can't 😒 . Thanks for all the help 🙏  I really appreciate it.
